### PR TITLE
US132983 - Convert `d2l-rubric-editable-score` to Lit

### PR DESCRIPTION
**Pull request by semibran**
[`2a09e1c`](2a09e1cf5d8b1cd23f8d2a980f2c38e35b0df887) - tried squashing with limited success. Some of these components just require too much internal refactoring for diffs to work cleanly.

### CODE REVIEW

File: src/components/rubric-editable-score.js [hunk 1]
```diff
@@ -0,0 +1,809 @@
...
+		} else {
+			return this.localize('outOf', 'outOf', entity.outOf.toString());
+		}
+	}
+
+	_localizeStarLabel() {
+		if (this.isCriterion) {
+			return this.localize('overriddenTotalScoreTip');
+		} else {
+			return this.localize('overriddenScoreTip');
```
Comment (MitchelPaulin) [inline src/components/rubric-editable-score.js:614]: Should these be flipped?

File: src/components/rubric-editable-score.js [hunk 2]
```diff
@@ -0,0 +1,809 @@
...
+		return {
+			token: { type: String },
+			criterionHref: { type: String },
+			entity: { type: Object },	// Entity can be a criterionEntity or a rubricEntity
+			assessmentHref: { type: String },
+			assessmentEntity: { type: Object },
+
+			editingScore: {
+				type: Number,
+				notify: true
```
Comment (MitchelPaulin) [inline src/components/rubric-editable-score.js:38]: Is `notify: true` still available in Lit?

File: src/components/rubric-editable-score.js [hunk 3]
```diff
@@ -0,0 +1,809 @@
...
+	}
+
+	get _isStaticView() {
+		return this.readOnly || !this.assessmentHref;
+	}
+
+	get _savedScore() {
+		return this.entity && this.assessmentEntity
+			? this.totalScore || this.assessmentEntity.score
+			: null;
```
Comment (MitchelPaulin) [inline src/components/rubric-editable-score.js:462]:
```suggestion
		if (this.entity) {
		    return this.totalScore || this.assessmentEntity?.score;
		}
```

File: src/components/rubric-editable-score.js (hunk 2 above)
Comment (semibran) [reply src/components/rubric-editable-score.js:38]: Nope, forgot to make a note to convert these to events

File: src/components/rubric-editable-score.js [hunk 4]
```diff
@@ -0,0 +1,809 @@
...
+				:host([overridden-styling]) #star {
+					display: inline-flex;
+				}
+
+				#clear-button {
+					margin-left: 0;
+					margin-right: 0.5rem;
+				}
+
+				:host-context([dir='rtl']) #clear-button {
```
Comment (imcdonald-d2l) [inline src/components/rubric-editable-score.js:164]: Looks like you're missing the `RtlMixin`

File: src/components/rubric-editable-score.js [hunk 5]
```diff
@@ -0,0 +1,809 @@
...
+		return html`
+			${!this.readOnly ? html`
+				${this._shouldShowClearCriterionOverrideButton('full') ? this._renderClearOverrideButton() : null}
+				${this._renderScoreInput()}
+			` : this._renderScoreLabel()}
+			<d2l-tooltip
+				align="start"
+				position="top"
+				aria-hidden="true"
+				?hidden="${!this._shouldShowTooltip}"
```
Comment (imcdonald-d2l) [inline src/components/rubric-editable-score.js:414]: This should be conditionally rendered to stay consistent with how we're writing components. That way we can remove this as well: 
```
[hidden] {
	display: none !important;
}
```

**Pull request merged**